docs/proposals/application-name-identifier.md

```diff
@@ -0,0 +1,397 @@
+---
+title: Change the way application resources are identified
+authors:
+  - "@jannfis"
+sponsors:
+  - TBD
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+creation-date: 2021-06-07
+last-updated: 2021-06-07
+---
+
+# Change the way application resources are identified
+
+This is a proposal to introduce using a hash value as application identifier
+in the application instance label. This will allow application names longer
+than 63 characters. As an additional goal, we propose to introduce a GUID as
+installation ID that will allow multiple Argo CD instances manage resources
+on the same cluster, without the need for manual reconfiguration.
+
+## Open Questions [optional]
+
+The major open questions are:
+
+* Which hashing algorithm to use? The authors think that `SHA-1` and `FNV-1a`
+  are the best options on the table, with `FNV-1a` being the more performant
+  one and `SHA-1` being the one most resilient against collisions with a
+  compromise for speed. There is a pretty good comparison of non cryptographic
+  hash algorithms on Stack Exchange that suggests that `FNV-1a` might be more
+  than sufficient for our use-cases:
+  https://softwareengineering.stackexchange.com/a/145633
+
+* The proposal defines a stretch goal for allowing multiple Argo CD instances
+  to manage applications on the same cluster, without the need for changing
+  the _application instance label key_ of each installation. This change may
+  sound intrusive, but the authors think the benefits would outweigh the
+  disadvantages. 
+
+## Summary
+
+Argo CD identifies resources it manages by setting the _application instance
+label_ to the name of the managing `Application` on all resources that are
+managed (i.e. reconciled from Git). The default label used is the well-known
+label `app.kubernetes.io/instance`.
+
+This proposal suggests to change the _value_ of this label to not use the
+literal application name, but instead use a value from a stable and collision
+free hash algorithm.
+
```

Comment:
A nit: the terminology here can be updated to 'virtually collision free' here, as I think no finite hash algorithm is truly collision free (and you are already making this distinction elsewhere in the proposal)

Reply:
Thanks, added _reasonable_ in front of _collision free_ :)